dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/MetaDataControllerTest.kt

```diff
@@ -328,17 +315,32 @@ class MetaDataControllerTest {
         return listOfActiveMetaData[0]
     }
 
+    @Suppress("kotlin:S138")
```

Comment:
No reason for this...
Either we adapt the rule, or we stick to it. Don't just add random supresses...

Reply:
I've had a chat about it with @SiegfriedSobkowiak. In our e2eapi tests the 20 line function rule tends to make the code less readable. E.g. we've factored out the asserts of a single test into a convoluted, super specific function e.g. here: https://github.com/d-fine/Dataland/blob/main/dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/MetaDataControllerTest.kt#L310

I generally do think that the 20 line rule should not apply to functions with @Test annotation as it leads to not testing the functionality directly but putting lots of helper functions between the test and the API and asserting less that the state at the end of the test is as expected to "safe LoCs".

Reply:
Here is the reasoning that Sonar gives on why long LoC functions should be avoided. In my opinion, most do not apply in the special case of an e2e test function.

![20LoCs](https://user-images.githubusercontent.com/35459207/226893243-1fe1bfa0-83ce-4f63-9b19-d8caebcbfeb7.png)

Reply:
``` 
    fun `ensure that version history field in metadata endpoint of meta data controller works`() {
        val companyId = apiAccessor.uploadOneCompanyWithRandomIdentifier().actualStoredCompany.companyId

        val frameworkDataAlpha = apiAccessor.testDataProviderForEuTaxonomyDataForNonFinancials.getTData(1)[0]
        val reportingPeriod = "2022"
        val firtUploadMetadata = apiAccessor.uploadWithWait(
                companyId,
                frameworkDataAlpha,
                reportingPeriod,
                apiAccessor.euTaxonomyNonFinancialsUploaderFunction,
        )
        val newNumberOfEmployees = (frameworkDataAlpha.numberOfEmployees ?: BigDecimal.ZERO) + BigDecimal.ONE
        val frameworkDataBeta = frameworkDataAlpha.copy(numberOfEmployees = newNumberOfEmployees)
        val secondUploadMetaData = apiAccessor.euTaxonomyNonFinancialsUploaderFunction(
                companyId,
                frameworkDataBeta,
                reportingPeriod,
        )
        apiAccessor.ensureQaIsPassed(listOf(firtUploadMetadata, secondUploadMetaData))
        val dataType = DataTypeEnum.eutaxonomyMinusNonMinusFinancials
        val activeDataset =
            apiAccessor.dataControllerApiForEuTaxonomyNonFinancials.getCompanyAssociatedEuTaxonomyDataForNonFinancials(
                validateShowActiveFlagAndReturnActiveDataMetaInfo(companyId, dataType, reportingPeriod, 2).dataId,
            )
        assertTrue(
            (activeDataset.data!!.numberOfEmployees == newNumberOfEmployees),
            "The active dataset does not have numberOfEmployees of the old one plus 1.",
        )
    }
```
This is 3 lines shorter - and is even more readable than the current solution.

Reply:
But in general: I am ok with your reasoning. But in this case, we should adopt the rules (e.g.: Don't apply kotlin:S138 to Tests).
We then should agree on this, and implement it. 
Or we should have a rule stating "It's ok to suppres kotlin:S138 in Tests" beacaus it's too much effort to make sonar comply to our wishes. 

But I think just - on an individual assessment - supressing team rules is not a good practice.

Reply:
resolved as per team discussion this morning